pkg/injector/bootstrap.go

```diff
@@ -95,9 +104,123 @@ func (wh *mutatingWebhook) marshalAndSaveBootstrap(name, namespace string, confi
 			bootstrap.EnvoyXDSCACertFile:                  cert.GetTrustedCAs(),
 			bootstrap.EnvoyXDSCertFile:                    cert.GetCertificateChain(),
 			bootstrap.EnvoyXDSKeyFile:                     cert.GetPrivateKey(),
+			signingIssuerIDKey:                            []byte(cert.GetSigningIssuerID()),
+			validatingIssuerIDKey:                         []byte(cert.GetValidatingIssuerID()),
 		},
 	}
 
 	log.Debug().Msgf("Creating bootstrap config for Envoy: name=%s, namespace=%s", name, namespace)
 	return wh.kubeClient.CoreV1().Secrets(namespace).Create(context.Background(), secret, metav1.CreateOptions{})
 }
+
+// NewBootstrapSecretRotator returns a new bootstrap secret rotator.
+func NewBootstrapSecretRotator(compIf compute.Interface, certManager *certificate.Manager, checkInterval time.Duration) *BootstrapSecretRotator {
+	return &BootstrapSecretRotator{
+		computeInterface: compIf,
+		certManager:      certManager,
+		checkInterval:    checkInterval,
+	}
+}
+
+// getBootstrapSecrets returns the bootstrap secrets stored in the informerCollection's store.
+func (b *BootstrapSecretRotator) getBootstrapSecrets() []*models.Secret {
+	secrets := b.computeInterface.ListSecrets()
+	var bootstrapSecrets []*models.Secret
+
+	for _, secret := range secrets {
+		// finds bootstrap secrets
+		if strings.Contains(secret.Name, bootstrapSecretPrefix) {
+			bootstrapSecrets = append(bootstrapSecrets, secret)
+		}
+	}
+	return bootstrapSecrets
+}
+
+// rotateBootstrapSecrets updates the bootstrap secret by getting the current or issuing a new certificate.
+func (b *BootstrapSecretRotator) rotateBootstrapSecrets(ctx context.Context) {
+	bootstrapSecrets := b.getBootstrapSecrets()
+	for _, secret := range bootstrapSecrets {
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			cert, err := getCertFromSecret(secret)
```

Comment:
I think we had added a `GetSecret()` method and call here. Why was this removed? Thanks!

Reply:
`GetSecret()` was essentially used to convert the bootstrap secret to a corev1.Secret, but we don't need to explicitly use the corev1.Secret, the models.Secret works here since we only need the secret data to get the cert

Reply:
I thought we added `GetSecret()` to ensure we had the latest version of the secret. Currently, if there is a conflict on a secret update, retry will always fail because the same outdated secret will be used to try to make the update.

Reply:
Oops you are right, I forgot that's how RetryOnConflict worked, will add it back